packages/astro/src/vite-plugin-head/index.ts

```diff
@@ -45,9 +45,34 @@ export default function configHeadVitePlugin(): vite.Plugin {
 
 	return {
 		name: 'astro:head-metadata',
+		enforce: 'pre',
+		apply: 'serve',
 		configureServer(_server) {
 			server = _server;
 		},
+		resolveId(source, importer) {
```

Comment:
If we're handling the propagation in `resolveId`. Could we remove the similar code in the `transform` hook entirely? Or do they work differently?

Reply:
I was thinking about this... I think we probably could. I don't fully trust out test coverage though. I'd rather leave a TODO and try and remove it later when feeling more brave.

Reply:
Works for me 👍 Probably worth circle back whenever possible as it touches on perf